packages/core/src/components/ion-route.tsx

```diff
@@ -53,25 +52,39 @@ export function IonRoute<TComponent extends string | ((...args: any[]) => any) =
 	props: IonRouteProps<TComponent>
 ) {
 	const router = useRouter();
-	const [_, propsToFix] = splitProps(props, ['component', 'componentProps']);
-	const fixedProps = () => fixProps(propsToFix);
+	const [_, restProps] = splitProps(props, [
+		'component',
+		'componentProps',
+		'beforeEnter',
+		'beforeLeave',
+		'url',
+	]);
+
+	const ionProps = () => ({
+		'prop:beforeEnter': props.beforeEnter,
+		'prop:beforeLeave': props.beforeLeave,
+		'prop:url': props.url,
+	});
+
+	const componentProps = () =>
+		typeof props.component === 'string'
+			? undefined
+			: {
+					component: RouteRouterContextProvider,
+					componentProps: {
+						component: props.component,
+						componentProps: props.componentProps ?? {},
+						router: router,
+					} satisfies RouteRouterContextProviderProps,
+				};
 
 	return (
 		<ion-route
-			{...fixedProps()}
+			{...ionProps()}
+			{...restProps}
+			// I keep this without `prop:` prefix bcs it's not required and at least we can see the value in the DOM
 			component={typeof props.component === 'string' ? props.component : ComponentRefElementName}
```

Comment:
I am not sure, but I think this is what broke the original router for me. Without it only registered webcomponent strings were supported. Need to double-check

Reply:
Nope, thats not it. I'm at a loss as to why the proxy webcomponent was needed :sweat_smile: